### PR TITLE
[DON'T MERGE] Basic playground

### PR DESCRIPTION
**Pull request by zpl-zak**

### CODE REVIEW

File: components/platform/input_manager.hpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+
+class InputManager
+{
+public:
+    virtual void init() = 0;
+    virtual void shutdown() = 0;
+    virtual void update() = 0;
+    
+    virtual bool getKeyPress() = 0;
```
Comment (drummyfish) [inline components/platform/input_manager.hpp]: Is this supposed to be expanded to `bool getKeyPressed(key)`?

File: components/platform/input_manager.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    virtual bool getKeyDown() = 0;
```
Comment (drummyfish) [inline components/platform/input_manager.hpp]: What is this supposed to do?

Comment (zaklaus) [reply components/platform/input_manager.hpp]: There are 3 recognized key states you can rely upon. getKeyPress(ed) returns true if key is being held down continously, useful when processing key input for movement and such. getKeyDown is supposed to be used for actions that aren't recurring that often (like jump, open inventory and such), getKeyUp tells us whether key was already released.

File: components/platform/input_manager.hpp (hunk 1 above)
Comment (zaklaus) [reply components/platform/input_manager.hpp]: Yes, this class is only a draft so far.

**Pull request closed without merging**